backend/tarsy/controllers/history_controller.py

```
@@ -240,8 +240,38 @@ async def get_session_detail(
+                            'step_description': event.get('step_description'),
+                            'duration_ms': event.get('duration_ms'),
+                            'details': event.get('details', {})
+                        })
+                
+                # Sort chronologically
+                stage_timeline.sort(key=lambda x: x['timestamp_us'])
+                
+                # Calculate interaction summary
+                llm_count = len([e for e in stage_timeline if e.get('type') == 'llm'])
+                mcp_count = len([e for e in stage_timeline if e.get('type') == 'mcp'])
+                total_duration = sum(e.get('duration_ms', 0) for e in stage_timeline if e.get('duration_ms'))
```

Ruff (0.12.2)

Line too long (109 > 88)

(E501)